web/src/services/api/storytellingApi.ts

```diff
@@ -0,0 +1,132 @@
+import { useMutation } from "@apollo/client";
+import { useCallback } from "react";
+
+import {
+  CreateStoryInput,
+  CreateStoryMutation,
+  CreateStoryPageInput,
+  CreateStoryPageMutation,
+  CreateTeamPayload,
+  DeleteStoryPageInput,
+  DeleteStoryPageMutation,
+  MoveStoryPageInput,
+  MoveStoryPageMutation,
+  MutationCreateStoryArgs,
+  MutationCreateStoryPageArgs,
+  MutationMoveStoryPageArgs,
+  MutationRemoveStoryPageArgs,
+} from "@reearth/services/gql/__gen__/graphql";
+import {
+  CREATE_STORY,
+  CREATE_STORY_PAGE,
+  DELETE_STORY_PAGE,
+  MOVE_STORY_PAGE,
+} from "@reearth/services/gql/queries/storytelling";
+import { useT } from "@reearth/services/i18n";
+
+import { useNotification } from "../state";
+
+export type Team = CreateTeamPayload["team"];
+
+export default function useStorytellingAPI() {
+  const t = useT();
+  const [, setNotification] = useNotification();
+
+  const [createStoryMutation] = useMutation<CreateStoryMutation, MutationCreateStoryArgs>(
+    CREATE_STORY,
+  );
+  const createStory = useCallback(
+    async (input: CreateStoryInput, opt?: { disableNotification?: boolean }) => {
+      const { data, errors } = await createStoryMutation({
+        variables: {
+          input,
+        },
+      });
+      if (errors || !data?.createStory?.story?.id) {
+        console.log("GraphQL: Failed to create story", errors);
+        if (!opt?.disableNotification) {
+          setNotification({ type: "error", text: t("Failed to create story.") });
+        }
+
+        return { status: "error", errors };
+      }
+
+      if (!opt?.disableNotification) {
+        setNotification({ type: "success", text: t("Successfully created story!") });
+      }
+      return { data, status: "success" };
+    },
+    [createStoryMutation, setNotification, t],
+  );
+
+  const [createStoryPageMutation] = useMutation<
+    CreateStoryPageMutation,
+    MutationCreateStoryPageArgs
+  >(CREATE_STORY_PAGE);
+
+  const createStoryPage = useCallback(
```

Comment:
We are using useXXXX to mirror apollo query and gql standards around their hooks (ie useQuery, useLazyQuery, useMutation, etc).
I get what you mean, and maybe in the future we will refactor them to more standard functional names (ie handleXxxXxx), but for now can you name the mutation and query callbacks to the same standard? 🙏🏻

Reply:
Also, there is a type `MutationReturn` for mutation callbacks. An example for project creation's return type is: `Promise<MutationReturn<Partial<Project>>>`. Please add that so we can make sure we are returning status correctly as well as the appropriate object

Reply:
Fixed in https://github.com/reearth/reearth/pull/592/commits/eeb17278c4f7968cb68359f6ba95860bd67567fc and https://github.com/reearth/reearth/pull/592/commits/0f7915e19785bc1a86ce38ba497755b00ac2e20a

Added errors to mutation return since we sometimes need it.
https://github.com/reearth/reearth/commit/0f7915e19785bc1a86ce38ba497755b00ac2e20a#diff-1d80e33480c523e4a6f8dcff37487486759567b683aa0dcab664f5d077f8b28cR11